FF8/Memory.cs

```diff
@@ -28,20 +25,64 @@ class Memory
 
         public static GameTime gameTime;
 
-        internal static ushort prevmusic = 0;
+        private static ushort prevmusic = 0;
         private static ushort currmusic = 0;
-        internal static ushort MusicIndex { get { return currmusic; } set { prevmusic = currmusic; currmusic = value; } }
+        internal static ushort MusicIndex
+        {
+            get
+            {
+                while ((prevmusic > currmusic || prevmusic == ushort.MinValue && currmusic == ushort.MaxValue) &&
+                    !dicMusic.ContainsKey(currmusic))
+                {
+
+                    if (dicMusic.Keys.Max() < currmusic)
+                    {
+                        currmusic = dicMusic.Keys.Max();
+                    }
+                    else
+                    {
+                        currmusic--;
+                    }
+                }
+                while (prevmusic < currmusic && !dicMusic.ContainsKey(currmusic))
+                {
+
+                    if (dicMusic.Keys.Max() < currmusic)
+                    {
+                        currmusic = dicMusic.Keys.Min();
+                    }
+                    else
+                    {
+                        currmusic++;
+                    }
+                }
+
+                return currmusic;
+            }
+            set
+            {
+                prevmusic = currmusic;
+                currmusic = value;
+            }
+        }
         public static string[] musices;
         public static Dictionary<ushort, List<string>> dicMusic = new Dictionary<ushort, List<string>>(); //ogg and sgt files have same 3 digit prefix.
         public static void SpriteBatchStartStencil()
-        => spriteBatch.Begin(SpriteSortMode.BackToFront, BlendState.Opaque, null, graphics.GraphicsDevice.DepthStencilState);
+        {
+            spriteBatch.Begin(SpriteSortMode.BackToFront, BlendState.Opaque, null, graphics.GraphicsDevice.DepthStencilState);
+        }
+
         public static void SpriteBatchStartAlpha()
-            => spriteBatch.Begin(SpriteSortMode.Deferred, BlendState.AlphaBlend);
+        {
+            spriteBatch.Begin(SpriteSortMode.Deferred, BlendState.AlphaBlend);
```

Comment:
is there any specific reason for changing lambda expression to method block? Readability or something else?

Reply:
I had setup format document in visual studio. I try to run that on the document i'm working on to fix the tabs and such. That is one of the things it's set to do right now. I can turn it off I think.

Reply:
Now that I think about it I might of done this manually. And I completely forgot about it. I'll change it back.